### PR TITLE
[ABW-3562] Third party deposits UI updates

### PR DESCRIPTION
**Pull request by matiasbzurovski**
Jira ticket: [ABW-3562](https://radixdlt.atlassian.net/browse/ABW-3562)

## Description
Updates the Third-party deposits flow as well as:
- Extends `PlainListRow` to render any view as accessory 
- Remove `PlainListRowCore.Context.general`: we will now always use `.settings` (except for `ToogleView` that uses `.toggle`).
- Update `ResourcesListView` & `PreferenceListView` to use `PlainListRow`, instead of implementing similar views themselves.
- Allow `Hint` to allow `.detail`, which uses orange like `.warning` but with smaller font and no icon.

## Before vs After

| Before | After |
| - | - |
| <video src=https://github.com/user-attachments/assets/8288cebf-31d7-47b6-91e6-043d6e6d31cf> | <video src=https://github.com/user-attachments/assets/7e5b7ebb-c505-472b-9ac1-3bfde0348621> |

[ABW-3562]: https://radixdlt.atlassian.net/browse/ABW-3562?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -114,8 +114,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/radixdlt/sargon",
       "state" : {
-        "revision" : "5b6d946221837aa6bed3b5ff725eb5fe4f844c6c",
-        "version" : "1.0.31"
+        "revision" : "b6342f791da0b29ca639b37abccdad04465501a1",
+        "version" : "1.0.29"
```
Comment (danvleju-rdx) [inline RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:118]: revert to 1.0.31?

File: RadixWallet/Core/DesignSystem/Components/Hint.swift [hunk 2]
```diff
@@ -64,29 +64,40 @@ public struct Hint: View, Equatable {
...
 			nil
 		case .error, .warning:
 			.error
 		}
 	}
+
+	var textStyle: TextStyle {
+		switch viewState.kind {
+		case .info, .error, .warning:
+			.body2HighImportance
```
Comment (danvleju-rdx) [inline RadixWallet/Core/DesignSystem/Components/Hint.swift:98]: Based on Zeplin it looks like there are situations where `info` and `error` should be `body2Regular` 🤔

File: RadixWallet/Features/AccountPreferencesFeature/Children/ThirdPartyDeposits/AddAssets+View.swift [hunk 3]
```diff
@@ -50,26 +50,25 @@ extension AddAsset {
...
+					VStack(spacing: .medium3) {
+						titleView(viewStore.mode.title)
+						instructionsView(viewStore.mode.instructions)
+
+						resourceAddressView(viewStore)
+						if case .allowDenyAssets = viewStore.mode {
+							depositListSelectionView(viewStore)
 						}
-						.padding([.horizontal, .bottom], .medium1)
+						addAssetButton(viewStore)
```
Comment (danvleju-rdx) [inline RadixWallet/Features/AccountPreferencesFeature/Children/ThirdPartyDeposits/AddAssets+View.swift:61]: Separator above?

File: RadixWallet/Core/DesignSystem/Components/Hint.swift (hunk 2 above)
Comment (danvleju-rdx) [reply RadixWallet/Core/DesignSystem/Components/Hint.swift:98]: [info1](https://zpl.io/ykQnLZG), [info2](https://zpl.io/KGWRWlA), [error1](https://zpl.io/B1lXM4K), [error2](https://zpl.io/1MJ33qE)

> after confirming with Aftab that we want them differently

right, maybe they should be consistent across all screens

**Review by danvleju-rdx — APPROVED** [review]

**Review by GhenadieVP — APPROVED** [review]

**Pull request merged**